src/c_wrapper.jl

```diff
@@ -370,7 +370,15 @@ function _unsafe_scs_solve(model::_ScsDataWrapper{S,T}) where {S,T}
         pointer(model.c),
     )
     for (key, value) in model.options
-        setproperty!(model.settings, key, value)
+        if value isa AbstractString
+            c_value = Base.cconvert(Cstring, value)
+            unsafe_value = Base.unsafe_convert(Cstring, c_value)
+            GC.@preserve c_value begin
```

Comment:
yeah, in general supporting something that will be non-trivially `cconvert`ed is hard. Suppose that I have `a::MyFancyString` such that `Base.cconvert(Cstring, a)` = String(a.some_data)` (newly constructed string). Then `unsafe_value = Base.unsafe_convert(Cstring,  c_value)` is basically a pointer to `c_value` and while `c_value` is preserved during `setproperty!` we will be left with freed pointer in `model.settings` after GC kicks.

We'd need to make sure that `model.options` holds data that will _not_ be non-trivially `cconvert`ed...

Reply:
just checked and for _anything but_ `String` `cconvert` allocates new `String` ;)

how about checking that `c_value === value` and throwing asking user to provide options with honest `Strings`?

Reply:
Oh. Yeah I didn't think about the pointer `model.settings` part. How about just
```Julia
if value isa AbstractString && !(value isa String)
    error("You must pass a `String` as the value for $(key)") 
end
```

Reply:
another option (which I like a bit less) is to replace the value in `model.options` by `Base.cconvert(Cstring, value)`.